### PR TITLE
Added adam optmizer

### PR DESCRIPTION
**Pull request by AstikaNehra**
chapter 20 of 4th edition

### CODE REVIEW

File: notebooks/chapter 20/adam optimizer.ipynb [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+    "    g_t = grad_func(theta_0)  #computes the gradient of the stochastic function\n",
+    "    m_t = beta_1*m_t + (1-beta_1)*g_t  #updates the moving averages of the gradient\n",
+    "    v_t = beta_2*v_t + (1-beta_2)*(g_t*g_t)  #updates the moving averages of the squared gradient\n",
+    "    m_cap = m_t/(1-(beta_1**t))  #calculates the bias-corrected estimates\n",
+    "    v_cap = v_t/(1-(beta_2**t))  #calculates the bias-corrected estimates\n",
+    "    theta_0_prev = theta_0\n",
+    "    theta_0 = theta_0 - (alpha*m_cap)/(math.sqrt(v_cap)+epsilon)  #updates the parameters\n",
+    "    if(theta_0 == theta_0_prev):  #checks if it is converged or not\n",
+    "        break"
+   ]
```
[the remark this answers is not in the document]
Comment (tirthasheshpatel) [reply notebooks/chapter 20/adam optimizer.ipynb]: Actually, you don't need to write your own version of ``adam``. It's already present [here](https://github.com/AstikaNehra/aima-python/blob/76c9a62995e7e5cfe3462d351066b282cd97e1cf/deep_learning4e.py#L215). Just use `psource` to show the required links of code and use it to demonstrate the concept.

[the remark this answers is not in the document]
Comment (tirthasheshpatel) [reply notebooks/chapter 20/adam optimizer.ipynb]: For the above comment, I think [this](https://github.com/aimacode/aima-python/blob/master/notebooks/chapter19/Optimizer%20and%20Backpropagation.ipynb) notebook will help.

[the remark this answers is not in the document]
Comment (AstikaNehra) [reply notebooks/chapter 20/adam optimizer.ipynb]: Okay sir, i would definitely work on the same. Thank you for the inputs.

[the remark this answers is not in the document]
Comment (tirthasheshpatel) [reply notebooks/chapter 20/adam optimizer.ipynb]: You can get started on the stuff I have suggested or wait for the maintainers to respond. BTW I am just a contributor like you!

**Pull request closed without merging**